ocaml-lsp-server/test/e2e/__tests__/textDocument-hover.test.ts

```diff
@@ -35,6 +35,40 @@ describe("textDocument/hover", () => {
     });
   });
 
+  it("returns type inferred under cursor in a formatted way", async () => {
+    languageServer = await LanguageServer.startAndInitialize();
+    await languageServer.sendNotification("textDocument/didOpen", {
+      textDocument: Types.TextDocumentItem.create(
+        "file:///test.ml",
+        "ocaml",
+        0,
+        "let f a b c d e f g h i = 1 + a + b + c + d + e + f + g + h + i\n",
+      ),
+    });
+
+    let result = await languageServer.sendRequest("textDocument/hover", {
+      textDocument: Types.TextDocumentIdentifier.create("file:///test.ml"),
+      position: Types.Position.create(0, 4),
+    });
+
+    expect(result).toMatchObject({
+      contents: { kind: "plaintext", value: outdent`int ->
+                                                    int ->
+                                                    int ->
+                                                    int ->
+                                                    int ->
+                                                    int ->
+                                                    int ->
+                                                    int ->
+                                                    int ->
+                                                    int` },
```

Comment:
@panglesd does this generated type look right? It seems like a lot of lines for a pretty simpler signature.

Reply:
The value 
``` ocaml
let f a b c d e f g h i = 1 + a + b + c + d + e + f + g + h + i
```
has type `int -> int -> int -> int -> int -> int -> int -> int -> int -> int`, which is 67 characters. But we send `ocamlformat_rpc` the option `("margin", "63")`, so it breaks every line. So this is the intended behaviour!
We could modify the value `f` to something like 
``` ocaml
let f (_: string option) (_: float -> string) (_: (string, char) result) = 1
```
 to get 
``` ocaml
string option ->
(float -> string) ->
(string, char) result ->
int
```
which is less lines.